conbench/hacks.py

```diff
@@ -1,49 +1,97 @@
-def get_case(tags, include_dataset=False):
-    case = [
+from typing import Dict, List
+
+CASE_KEYS_DO_NOT_USE_FOR_NAME = ("id", "name", "suite", "source", "dataset")
+
+
+def get_case_kvpair_strings(tags: Dict[str, str], include_dataset=False) -> List[str]:
+    """
+    Build and and return a list of strings, each item reflecting a key/value
+    pair, all of which together define the "case permutation" of this
+    conceptual benchmark.
+
+    Each item takes the shape key=value.
+
+    Special keys are omitted (does that make sense?)
+
+    Keep those with None value.
+    """
+    kvpairs = [
         (k, v)
         for k, v in sorted(tags.items())
-        if v is not None
-        and k != "id"
-        and k != "name"
-        and k != "suite"
-        and k != "source"
-        and k != "dataset"
-        and k != "language"
+        if k not in CASE_KEYS_DO_NOT_USE_FOR_NAME
     ]
+
+    # Rationale?
     if include_dataset and "dataset" in tags:
-        case.append(("dataset", tags["dataset"]))
-    if "language" in tags:
-        case.append(("language", tags["language"]))
-    booleans = [True, False, "true", "false"]
-    case = [(k, f"{k}={v}") if v in booleans else (k, v) for k, v in case]
-    return [f"{k}={v}" if isinstance(v, (int, float)) else str(v) for k, v in case]
-
-
-def set_display_name(benchmark):
-    is_api = isinstance(benchmark, dict)
-    tags = benchmark["tags"] if is_api else benchmark.case.tags
-    name = tags.get("name") if is_api else benchmark.case.name
-    if "name" not in tags:
-        tags["name"] = name
-    case = get_case(tags, include_dataset=True)
-    if "suite" in tags:
-        case = [name] + case
-    name = ", ".join(case) if case else name
+        kvpairs.append(("dataset", tags["dataset"]))
+
+    # # Rationale?
+    # if "language" in tags:
+    #     kvpairs.append(("language", tags["language"]))
+
+    # booleans = [True, False, "true", "false"]
+
+    # case = [(k, f"{k}={v}") if v in booleans else (k, v) for k, v in case]
+
+    # return [f"{k}={v}" if isinstance(v, (int, float)) else str(v) for k, v in case]
+
+    return [f"{k}={v}" for k, v in kvpairs]
+
+
+def set_display_case_permutation(bmresult):
+    """
+    Build and set a string reflecting the case permutation for this bmresult
+    result.
+    """
+    is_api = isinstance(bmresult, dict)
+    tags = bmresult["tags"] if is_api else bmresult.case.tags
+
+    # Note(JP): this tries to pull in the benchmark name into the case
+    # permutation k/v pairs. Skip this for now. Unique combination must be
+    # name+case_perm, so adding the name into the case_perm again should not be
+    # conceptually needed.
+
+    # name = tags.get("name") if is_api else bmresult.case.name
+    # if "name" not in tags:
+    #     tags["name"] = name
```

Comment:
This added `name` back into the output. But I think it might be cleaner to not have it here on the output side of things. It always _is_ contained in the output, as higher-level property (the benchmark name, previously "batch name").